controller/test/units/units_test.cpp

```diff
@@ -0,0 +1,187 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "units.h"
+#include "gtest/gtest.h"
+
+TEST(Units, Pressure) {
+  EXPECT_FLOAT_EQ(kPa(1).kPa(), 1);
+  EXPECT_FLOAT_EQ(kPa(1).cmH2O(), 10.1972);
+  EXPECT_FLOAT_EQ(cmH2O(1).cmH2O(), 1);
+  EXPECT_FLOAT_EQ((kPa(1) - kPa(2)).kPa(), -1);
+  EXPECT_FLOAT_EQ((cmH2O(1) + cmH2O(10)).cmH2O(), 11);
+  EXPECT_FLOAT_EQ((kPa(1) - cmH2O(10.1972)).kPa(), 0);
+
+  EXPECT_TRUE(kPa(1) < kPa(2));
+  EXPECT_FALSE(kPa(1) < kPa(1));
+  EXPECT_FALSE(kPa(2) < kPa(1));
+
+  EXPECT_TRUE(kPa(1) <= kPa(2));
+  EXPECT_TRUE(kPa(1) <= kPa(1));
+  EXPECT_FALSE(kPa(2) <= kPa(1));
+
+  EXPECT_FALSE(kPa(1) == kPa(2));
+  EXPECT_TRUE(kPa(1) == kPa(1));
+
+  EXPECT_TRUE(kPa(1) != kPa(2));
+  EXPECT_FALSE(kPa(1) != kPa(1));
+
+  EXPECT_FALSE(kPa(1) >= kPa(2));
+  EXPECT_TRUE(kPa(1) >= kPa(1));
+  EXPECT_TRUE(kPa(2) >= kPa(1));
+
+  EXPECT_FALSE(kPa(1) > kPa(2));
+  EXPECT_FALSE(kPa(1) > kPa(1));
+  EXPECT_TRUE(kPa(2) > kPa(1));
+}
+
+TEST(Units, Length) {
+  EXPECT_FLOAT_EQ(meters(1).meters(), 1);
+  EXPECT_FLOAT_EQ(meters(1).millimeters(), 1000);
+  EXPECT_FLOAT_EQ(millimeters(1).millimeters(), 1);
+  EXPECT_FLOAT_EQ(millimeters(1).meters(), 0.001);
+  EXPECT_FLOAT_EQ((meters(1) - meters(2)).meters(), -1);
+  EXPECT_FLOAT_EQ((millimeters(1) + millimeters(10)).millimeters(), 11);
+  EXPECT_FLOAT_EQ((meters(1) - millimeters(1000)).meters(), 0);
+
+  EXPECT_TRUE(meters(1) < meters(2));
```

Comment:
This stuff repeats with other units, worth extracting into a template function `template <typename T> ExpectComparisonWellBehaved(T lesser, T greater)`?

Reply:
Much better, thanks.